### PR TITLE
feat: Delete given Thunderstore mod

### PR DESCRIPTION
**Pull request by GeckoEidechse**
Deletes a given Thunderstore Mod.

- [x] #109
- [x] #110
- [x] Hook up delete mod button on Thunderstore view

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
@Alystrasz in regards to UI, how should we best add it to the ThunderstoreView?

- Add an additional trashcan icon for installed mods?
- Hovering "_Installed_" changes the button to "_Removing_" (doesn't work with outdated mods)

Other ideas? ^^

**Comment by Alystrasz** [comment]
I would replace the information button with a [MoreFields icon button](https://element-plus.org/en-US/component/icon.html#icon-collection), which when clicked, reveals a [dropdown menu](https://element-plus.org/en-US/component/dropdown.html) with the following options:
* More info (open Thunderstore mod page)
* Remove mod

File: src-vue/src/components/ThunderstoreModCard.vue [hunk 1]
```diff
@@ -35,11 +35,30 @@
...
+                <el-dropdown v-else>
+                    <el-icon class="infoBtn moreBtn">
+                        <MoreFilled />
+                    </el-icon>
+                    <template #dropdown>
+                        <el-dropdown-menu>
+                            <el-dropdown-item @click="openURL(mod.package_url)">
+                                More info
+                            </el-dropdown-item>
+                            <el-dropdown-item  @click="deleteMod(mod)">
```
Comment (Alystrasz) [inline src-vue/src/components/ThunderstoreModCard.vue:56]: Don't you think an approval prompt is required here, just like local mods?

File: src-tauri/src/mod_management/mod.rs [hunk 2]
```diff
@@ -380,3 +385,88 @@ pub async fn fc_download_mod_and_install(
...
+    // Get folder name based on Thundestore mod string
+    for installed_ns_mod in installed_ns_mods {
+
+        if installed_ns_mod.thunderstore_mod_string.is_none() {
+            // Not a Thunderstore mod
+            continue;
+        }
+
+        // Installed mod matches specified Thunderstore mod string
+        if installed_ns_mod.thunderstore_mod_string.unwrap() == thunderstore_mod_string {
```
[the remark this answers is not in the document]
Comment (GeckoEidechse) [reply src-tauri/src/mod_management/mod.rs]: Already had it finished, forgot to commit. Whoops.

Done in 0bc52918fa4963cc7c8daa66e566322b6bc02841

File: src-vue/src/components/ThunderstoreModCard.vue [hunk 3]
```diff
@@ -166,6 +166,34 @@ export default defineComponent({
             return `${dependencyStringMembers[0]}-${dependencyStringMembers[1]}`;
         },
 
+        async deleteMod(mod: ThunderstoreMod) {
+            let game_install = {
+                game_path: this.$store.state.game_path,
+                install_type: this.$store.state.install_type
+            } as GameInstall;
+
+            await invoke("delete_thunderstore_mod_caller", { gameInstall: game_install, thunderstoreModString: this.latestVersion.full_name })
```
[the remark this answers is not in the document]
Comment (GeckoEidechse) [reply src-vue/src/components/ThunderstoreModCard.vue]: Resolved in 0bc52918fa4963cc7c8daa66e566322b6bc02841 by only comparing author and mod name.

File: src-vue/src/components/ThunderstoreModCard.vue (hunk 1 above)
Comment (GeckoEidechse) [reply src-vue/src/components/ThunderstoreModCard.vue:56]: Didn't you write that frontend code? :P

Anyway, it tried doing like 
```html
<el-dropdown-item>
    <el-popconfirm title="Are you sure to delete this mod?" @confirm="deleteMod(mod)">
        <template #reference>
            Remove mod  
        </template>
    </el-popconfirm>
</el-dropdown-item>
```

but that cause the dropdown menu to vanish the moment I hover the pop-up menu.

Demo:

https://user-images.githubusercontent.com/40122905/211149596-ee047a56-9d02-4c95-91c1-3ad490ef767f.mp4

Any ideas on how to resolve? ^^

File: src-tauri/src/mod_management/mod.rs [hunk 4]
```diff
@@ -430,3 +430,50 @@ pub fn delete_northstar_mod(game_install: GameInstall, nsmod_name: String) -> Re
...
+            // Add folder to list of folder to remove
+            mod_folders_to_remove.push(installed_ns_mod.directory);
+        }
+    }
+
+    if !(mod_folders_to_remove.len() > 0) {
+        return Err(format!(
+            "No Northstar mod matching {thunderstore_mod_string} found to be installed"
```
[the remark this answers is not in the document]
Comment (GeckoEidechse) [reply src-tauri/src/mod_management/mod.rs]: I updated the phrasing in acf1ddc28e69d5b90a03cafcfdf971b11228ec3c to a mixture of both ^^

File: src-vue/src/components/ThunderstoreModCard.vue (hunk 1 above)
Comment (Alystrasz) [reply src-vue/src/components/ThunderstoreModCard.vue:56]: > Didn't you write that frontend code? :P

Hey, I can't do all your work 😄 

If popconfirm is not compatible with dropdown item, you can use a confirm message box: 
https://element-plus.org/en-US/component/message-box.html#confirm

Comment (GeckoEidechse) [reply src-vue/src/components/ThunderstoreModCard.vue:56]: Okay, that worked ^^
Implemented in 16dd15ff7f9cebc82b9800c653c0d7b59a9ef2e9

Demo:
https://user-images.githubusercontent.com/40122905/211193481-79f84f51-6fae-4c52-9f5b-a45afd570db8.mp4

Comment (Alystrasz) [reply src-vue/src/components/ThunderstoreModCard.vue:56]: Looks dope!
Maybe we should use this and [Message API](https://element-plus.org/en-US/component/message.html) more often in FlightCore :)

**Review by Alystrasz — APPROVED** [review]
Confirmed working on Windows.

File: src-vue/src/components/ThunderstoreModCard.vue (hunk 1 above)
Comment (GeckoEidechse) [reply src-vue/src/components/ThunderstoreModCard.vue:56]: Definitely ^^

**Pull request merged**